corehq/apps/case_search/xpath_functions/ancestor_functions.py

```diff
@@ -144,11 +144,21 @@ def _get_case_ids_from_ast_filter(context, filter_node):
         es_filter = build_filter_from_ast(filter_node, context)
 
         es_query = CaseSearchES().domain(context.domain).filter(es_filter)
-        if es_query.count() > MAX_RELATED_CASES:
-            new_query = serialize(filter_node)
-            raise TooManyRelatedCasesError(
-                gettext("The related case lookup you are trying to perform would return too many cases"),
-                new_query
-            )
 
         return es_query.scroll_ids()
+
+
+def is_ancestor_query_below_limit(query, xpath):
+    node = parse(xpath)
+    REDUCED_MAX_RELATED_CASES = 5000  # for testing purposes
+    if (is_ancestor_comparison(node) or 'ancestor-exists' in xpath) \
+       and query.count() > REDUCED_MAX_RELATED_CASES:
+        try:
+            query = BinaryExpression(node.left.right, node.op, node.right)
+        except AttributeError:
+            ancestor_path_node, ancestor_case_filter_node = node.args
+            query = ancestor_case_filter_node
```

Comment:
Just for clarification, is how we're understanding this correct? 
If the xpath argument is "color = "red"", "red" would be saved as `ancestor_case_filter_node` and also as `query` which is what's printed as the error?

Reply:
sorry for the delay but no. For ancestor functions, `ancestor_case_filter_node` would be the full "color = "red" argument since its the right side of the full search term parent/color = "red".